$NetBSD$

Build with the common Git xdiff source and libgit2 adapter.

--- deps/xdiff/CMakeLists.txt.orig
+++ deps/xdiff/CMakeLists.txt
@@ -1,28 +1,20 @@
-
-file(GLOB SRC_XDIFF "*.c" "*.h")
-list(SORT SRC_XDIFF)
+set(SRC_XDIFF
+	"${XDIFF_SOURCE_DIR}/xdiffi.c"
+	"${XDIFF_SOURCE_DIR}/xemit.c"
+	"${XDIFF_SOURCE_DIR}/xhistogram.c"
+	"${XDIFF_SOURCE_DIR}/xmerge.c"
+	"${XDIFF_SOURCE_DIR}/xpatience.c"
+	"${XDIFF_SOURCE_DIR}/xprepare.c"
+	"${XDIFF_SOURCE_DIR}/xutils.c")
 
 add_library(xdiff OBJECT ${SRC_XDIFF})
+set_target_properties(xdiff PROPERTIES C_STANDARD 99 C_STANDARD_REQUIRED ON)
+target_compile_definitions(xdiff PRIVATE XDL_COMPAT_HEADER=\"git-xdiff.h\")
 target_include_directories(xdiff SYSTEM PRIVATE
+	"${XDIFF_SOURCE_DIR}"
+	"${PROJECT_SOURCE_DIR}/deps/xdiff"
 	"${PROJECT_SOURCE_DIR}/include"
 	"${PROJECT_SOURCE_DIR}/src/util"
 	"${PROJECT_BINARY_DIR}/gen_headers"
 	${LIBGIT2_SYSTEM_INCLUDES}
 	${LIBGIT2_DEPENDENCY_INCLUDES})
-
-# the xdiff dependency is not (yet) warning-free, disable warnings
-# as errors for the xdiff sources until we've sorted them out
-if(MSVC)
-        set_source_files_properties(xdiffi.c PROPERTIES COMPILE_FLAGS -WX-)
-        set_source_files_properties(xemit.c PROPERTIES COMPILE_FLAGS -WX-)
-        set_source_files_properties(xhistogram.c PROPERTIES COMPILE_FLAGS -WX-)
-        set_source_files_properties(xmerge.c PROPERTIES COMPILE_FLAGS -WX-)
-        set_source_files_properties(xutils.c PROPERTIES COMPILE_FLAGS -WX-)
-        set_source_files_properties(xpatience.c PROPERTIES COMPILE_FLAGS -WX-)
-else()
-        set_source_files_properties(xdiffi.c PROPERTIES COMPILE_FLAGS "-Wno-sign-compare -Wno-unused-parameter")
-        set_source_files_properties(xemit.c PROPERTIES COMPILE_FLAGS "-Wno-sign-compare -Wno-unused-parameter")
-        set_source_files_properties(xhistogram.c PROPERTIES COMPILE_FLAGS "-Wno-sign-compare")
-        set_source_files_properties(xutils.c PROPERTIES COMPILE_FLAGS "-Wno-sign-compare")
-        set_source_files_properties(xpatience.c PROPERTIES COMPILE_FLAGS "-Wno-sign-compare")
-endif()
